pkg/platform/aws/aws.go

```diff
@@ -47,126 +43,99 @@ type workerPool struct {
 	Tags         map[string]string `hcl:"tags,optional"`
 }
 
-type config struct {
-	AssetDir                 string            `hcl:"asset_dir"`
-	ClusterName              string            `hcl:"cluster_name"`
-	Tags                     map[string]string `hcl:"tags,optional"`
-	OSName                   string            `hcl:"os_name,optional"`
-	OSChannel                string            `hcl:"os_channel,optional"`
-	OSVersion                string            `hcl:"os_version,optional"`
-	DNSZone                  string            `hcl:"dns_zone"`
-	DNSZoneID                string            `hcl:"dns_zone_id"`
-	ExposeNodePorts          bool              `hcl:"expose_nodeports,optional"`
-	SSHPubKeys               []string          `hcl:"ssh_pubkeys"`
-	CredsPath                string            `hcl:"creds_path,optional"`
-	ControllerCount          int               `hcl:"controller_count,optional"`
-	ControllerType           string            `hcl:"controller_type,optional"`
-	ControllerCLCSnippets    []string          `hcl:"controller_clc_snippets,optional"`
-	Region                   string            `hcl:"region,optional"`
-	EnableAggregation        bool              `hcl:"enable_aggregation,optional"`
-	DiskSize                 int               `hcl:"disk_size,optional"`
-	DiskType                 string            `hcl:"disk_type,optional"`
-	DiskIOPS                 int               `hcl:"disk_iops,optional"`
-	NetworkMTU               int               `hcl:"network_mtu,optional"`
-	HostCIDR                 string            `hcl:"host_cidr,optional"`
-	PodCIDR                  string            `hcl:"pod_cidr,optional"`
-	ServiceCIDR              string            `hcl:"service_cidr,optional"`
-	ClusterDomainSuffix      string            `hcl:"cluster_domain_suffix,optional"`
-	EnableReporting          bool              `hcl:"enable_reporting,optional"`
-	CertsValidityPeriodHours int               `hcl:"certs_validity_period_hours,optional"`
-	WorkerPools              []workerPool      `hcl:"worker_pool,block"`
+type flatcar struct {
+	OSName string `hcl:"os_name,optional"`
 }
 
-// init registers aws as a platform
-func init() {
-	platform.Register("aws", NewConfig())
+type network struct {
+	HostCIDR string `hcl:"host_cidr"`
```

Comment:
In a way, we use the same in Packet, except here we use it for setting the CIDR and for Packet to set the firewall. I think that should be moved to common fields.